File: src/main/java/org/kite9/diagram/visualization/compaction/align/Aligner.java
package org.kite9.diagram.visualization.compaction.align;

import java.util.List;

import org.kite9.diagram.model.AlignedRectangular;
import org.kite9.diagram.visualization.compaction.Compaction;

public interface Aligner {
	
	/**
	 * Applies alignment to all rectangulars along a the same given axis.
	 * Each AlignedRectangular may have a different alignment, but they will have all passed 
	 * through willAlign with true.
	 */
	void alignRectangulars(List<AlignedRectangular> de, Compaction c, boolean horizontal);
	
	boolean willAlign(AlignedRectangular de, boolean horizontal);
}

File: src/main/kotlin/org/kite9/diagram/visualization/compaction/align/Aligner.kt
package org.kite9.diagram.visualization.compaction.align

import org.kite9.diagram.model.Container
import org.kite9.diagram.model.Rectangular
import org.kite9.diagram.visualization.compaction.Compaction

interface Aligner {

    /**
     * Applies alignment to all rectangulars along a the same given axis.
     * Each AlignedRectangular may have a different alignment, but they will have all passed
     * through willAlign with true.
     */
    fun alignFor(co: Container, de: Set<Rectangular>, c: Compaction, horizontal: Boolean)
    fun willAlign(de: Rectangular, horizontal: Boolean): Boolean
}